### DISCUSSION TITLE
I get this error while I try to install teleport in a EC2 instance

### DISCUSSION THREAD

**Discussion by janudabethmin** [gravitational/teleport#50709] [Q&A]
```  
[root@ip-10-10-1-181 ~]# curl https://cdn.teleport.dev/install-v16.4.12.sh | bash -s 16.4.12
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 12099  100 12099    0     0   249k      0 --:--:-- --:--:-- --:--:--  246k
Installing Teleport v16.4.12 through dnf
Repository teleport is listed more than once in the configuration
Last metadata expiration check: 0:02:23 ago on Fri 20 Dec 2024 06:52:37 AM UTC.
Package dnf-plugins-core-4.3.0-13.amzn2023.0.5.noarch is already installed.
Dependencies resolved.
Nothing to do.
Complete!
Repository teleport is listed more than once in the configuration
Adding repo from: https://yum.releases.teleport.dev/amzn/2023/Teleport/x86_64/stable/v16/teleport-yum.repo
Repository teleport is listed more than once in the configuration
Last metadata expiration check: 0:02:23 ago on Fri 20 Dec 2024 06:52:37 AM UTC.
No match for argument: teleport-16.4.12
Error: Unable to find a match: teleport-16.4.12
```

**Comment by janudabethmin** [comment]
Can someone help me with this issue

**Comment by marcoandredinis** [comment]
Thank you for the report.
Can you please confirm that you are running this script in a Amazon Linux 2023 EC2 Instance?

I just tried that command on a brand new instance and was able to install teleport.

`Repository teleport is listed more than once in the configuration`

My best guess to what happened is that there's another repo offering the teleport package and that repo does not have the 16.4.12 version.


Can you please run the following command and post the output?
```
$ grep teleport /etc/yum.repos.d/*
```

**Comment by janudabethmin** [comment]
Hi, thank you for the response. @marcoandredinis 

Yes, it's an Amazon Linux ec2 2023. So I tried to connect to one instance using my teleport free trial account, and then self-host teleport there. So, the teleport.yaml file is already in use I think., and it's the cause of this error. Could we possibly do that? 

I want to do that because in my subnet all the ssh is already denied by a NACL. So, I spin up the instance and connect to teleport via a User Script.

The other thing I wanna know is would we be able to self-host teleport in a private instance and route traffic through an ELB. For now, I have done it using a public instance in another subnet. I want to do it in the same ssh denied subnet if I can.🤔

**Reply by marcoandredinis** [reply]
> So I tried to connect to one instance using my teleport free trial account, and then self-host teleport there

So, it probably ended up installing two repositories which might be tied to a specific major release.
Can you please post the output of `$ grep teleport /etc/yum.repos.d/*`?
It should also refuse to install another `/etc/teleport.yaml` file because it detects one already.

You can self-host teleport with an ELB in front of it 👍 
Here's a guide explaining how to do so
https://goteleport.com/docs/ver/17.x/admin-guides/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform/
